src/main/java/com/google/api/codegen/config/SampleSpec.java

```diff
@@ -0,0 +1,142 @@
+/* Copyright 2018 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.api.codegen.config;
+
+import com.google.api.codegen.MethodConfigProto;
+import com.google.api.codegen.SampleConfiguration;
+import com.google.api.codegen.SampleConfiguration.SampleTypeConfiguration;
+import com.google.api.codegen.SampleValueSet;
+import com.google.api.codegen.viewmodel.ClientMethodType;
+import com.google.common.collect.Iterables;
+import java.util.HashMap;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * Class SampleSpec stores the sample specification for a given method, and provides methods to
+ * easily access them by calling form and sample type.
+ */
+public class SampleSpec {
+
+  /** A reference to the Sample Configuration from which the other fields are derived. */
+  private final SampleConfiguration sampleConfiguration;
+
+  /** All the SampleValueSets defined for this method, indexed by their IDs. */
+  private Map<String, SampleValueSet> valueSets;
+
+  private boolean specified;
+
+  /** The various types of supported samples. */
+  public enum SampleType {
+    IN_CODE,
+    STANDALONE,
+    EXPLORER,
+  }
+
+  public SampleSpec(MethodConfigProto methodConfigProto) {
+    specified = methodConfigProto.hasSamples();
+    this.sampleConfiguration = methodConfigProto.getSamples();
```

Comment:
Be consistent with using `this.` vs not using it

Reply:
Must have been cruft. I try to use `this` only to resolve ambiguity.